### PR TITLE
APP-169 Android Sample "Cell monitor"

### PR DESCRIPTION
**Pull request by maxml**
http://jira.kaaproject.org/browse/APP-169

Blocked by http://jira.kaaproject.org/browse/APP-148 - not integrated in sandbox.
Code is working, I am working on integration process.

Please, review code.

### CODE REVIEW

**Comment by rasendubi** [comment]
Please, rebase on `develop`.

**Comment by maxml** [comment]
Rebased

**Comment by rasendubi** [comment]
Don't commit build directories or any auto-generated content.

**Comment by maxml** [comment]
Deleted

File: cellmonitor/source/android/CellMonitor/app/src/main/java/org/kaaproject/kaa/demo/cellmonitor/util/LocationUtil.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        Intent intent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
+                        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+                        context.startActivity(intent);
+                    }
+                });
+    }
+}
```
Comment (rasendubi) [inline cellmonitor/source/android/CellMonitor/app/src/main/java/org/kaaproject/kaa/demo/cellmonitor/util/LocationUtil.java]: No newline at end of file.

File: cellmonitor/source/android/CellMonitor/app/src/androidTest/java/org/kaaproject/kaa/demo/cellmonitor/ApplicationTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/**
```
Comment (rasendubi) [inline cellmonitor/source/android/CellMonitor/app/src/androidTest/java/org/kaaproject/kaa/demo/cellmonitor/ApplicationTest.java]: It should be a normal comment, not javadoc one.

File: cellmonitor/source/android/CellMonitor/gradlew [hunk 3]
```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+
+##############################################################################
+## Copyright 2014-2016 CyberVision, Inc.
```
Comment (rasendubi) [inline cellmonitor/source/android/CellMonitor/gradlew]: This file is not written by CyberVision, so our license don't apply.

**Comment by maxml** [comment]
Fixed

**Comment by maxml** [comment]
Fixed

File: cellmonitor/source/android/app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -30,6 +29,14 @@
...
     <string name="logs_sent">Sent log count:</string>
+
     <string name="unavailable">Unavailable</string>
 
+    <string name="no_internet_connection">You have no internet connection</string>
+    <string name="location_value">[ %1$s, %2$s ]</string>
+    <string name="logs_collected">Collected log count</string>
+    <string name="enable_gps">Enable GPS</string>
+    <string name="location_dialog_title">Turn on location services to determine your location</string>
+    <string name="network_dialog_title">Turn on network to allow working with server</string>
```
Comment (ikulikov) [inline cellmonitor/source/android/app/src/main/res/values/strings.xml]: Enable network in order to communicate with server

File: cellmonitor/source/android/app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -30,6 +29,14 @@
...
     <string name="last_log_time">Last log time:</string>
     <string name="logs_sent">Sent log count:</string>
+
     <string name="unavailable">Unavailable</string>
 
+    <string name="no_internet_connection">You have no internet connection</string>
+    <string name="location_value">[ %1$s, %2$s ]</string>
+    <string name="logs_collected">Collected log count</string>
+    <string name="enable_gps">Enable GPS</string>
+    <string name="location_dialog_title">Turn on location services to determine your location</string>
```
Comment (ikulikov) [inline cellmonitor/source/android/app/src/main/res/values/strings.xml]: Enable location services in order to detect your location

File: cellmonitor/source/android/app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -30,6 +29,14 @@
     <string name="gps_location">Phone GPS location:</string>
     <string name="last_log_time">Last log time:</string>
     <string name="logs_sent">Sent log count:</string>
+
     <string name="unavailable">Unavailable</string>
 
+    <string name="no_internet_connection">You have no internet connection</string>
```
Comment (ikulikov) [inline cellmonitor/source/android/app/src/main/res/values/strings.xml]: No internet access available

**Comment by ikulikov** [comment]
LGTM

File: cellmonitor/source/android/.gitignore [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+##############################################################################
+
+# Created by https://www.gitignore.io/api/android,intellij
+
+### Android ###
+# Built application files
+*.apk
```
Comment (ashvayka) [inline cellmonitor/source/android/.gitignore]: Are you sure that this .gitignore file should be here? Maybe one file for all projects is better and easier to maintain?

**Comment by ashvayka** [comment]
how was this tested? 
Sandbox already supports gradle plugins/build?

**Comment by maxml** [comment]
Yes, I already do it

**Comment by vadimol** [comment]
please, resolve merge conflicts locally and update the PR

**Pull request merged**